components/Common/Badge/index.tsx

```diff
@@ -0,0 +1,55 @@
+import ArrowRightIcon from '@heroicons/react/24/solid/ArrowRightIcon';
+import Link from 'next/link';
+import type { ComponentProps, FC, PropsWithChildren } from 'react';
+import styles from './index.module.scss';
+import { component } from '@/types/component';
+import classNames from 'classnames';
+
+type BadgeProps = {
+  kind?: component;
+  badgeText?: string;
+} & ComponentProps<typeof Link>;
+
+const Badge: FC<PropsWithChildren<BadgeProps>> = ({
+  kind,
+  badgeText,
+  children,
+  ...args
+}) => {
+  if (typeof kind === 'undefined') kind = component.Default;
+  const iconClasses = classNames(styles.icon, {
+    [styles.iconWarning]: kind === component.Warning,
+    [styles.iconError]: kind === component.Error,
+    [styles.iconGreen]: kind === component.Default,
+  });
+
+  const wrapperClasses = classNames(styles.wrapper, {
+    [styles.wrapperWarning]: kind === component.Warning,
+    [styles.wrapperError]: kind === component.Error,
+    [styles.wrapperGreen]: kind === component.Default,
+  });
+
+  const badgeClasses = classNames(styles.badge, {
+    [styles.badgeWarning]: kind === component.Warning,
+    [styles.badgeError]: kind === component.Error,
+    [styles.badgeGreen]: kind === component.Default,
+  });
+
+  const contentClasses = classNames(styles.content, {
+    [styles.contentWarning]: kind === component.Warning,
+    [styles.contentError]: kind === component.Error,
+    [styles.contentGreen]: kind === component.Default,
+  });
+
+  return (
+    <Link {...args}>
+      <div className={wrapperClasses}>
```

Comment:
This is against accessibility standards. An a element should not have a div inside. This would fail a11y.

It should be the other way around. Also you could apply the styles directly to the Link component. So you don't need to create a div.